cloud/services/loadbalancers.go

```diff
@@ -102,11 +107,129 @@ func CreateNodeBalancerConfig(ctx context.Context, clusterScope *scope.ClusterSc
 		Check:     linodego.CheckConnection,
 	}
 
-	if linodeNBConfig, err = clusterScope.LinodeClient.CreateNodeBalancerConfig(ctx, clusterScope.LinodeCluster.Spec.Network.NodeBalancerID, createConfig); err != nil {
+	if linodeNBConfig, err = clusterScope.LinodeClient.CreateNodeBalancerConfig(
+		ctx,
+		clusterScope.LinodeCluster.Spec.Network.NodeBalancerID,
+		createConfig,
+	); err != nil {
 		logger.Info("Failed to create Linode NodeBalancer config", "error", err.Error())
 
 		return nil, err
 	}
 
 	return linodeNBConfig, nil
 }
+
+// GetNodeBalancerConfig gets the first NodeBalancer config
+func GetNodeBalancerConfig(
+	ctx context.Context,
+	clusterScope *scope.ClusterScope,
+	logger logr.Logger,
+) (*linodego.NodeBalancerConfig, error) {
+	var linodeNBConfigs []linodego.NodeBalancerConfig
+	var err error
+
+	if linodeNBConfigs, err = clusterScope.LinodeClient.ListNodeBalancerConfigs(
+		ctx,
+		clusterScope.LinodeCluster.Spec.Network.NodeBalancerID,
+		linodego.NewListOptions(1, ""),
+	); err != nil {
+		logger.Info("Failed to list NodeBalancer Configs", "error", err.Error())
+
+		return nil, err
+	}
+	if len(linodeNBConfigs) == 0 {
+		return nil, err
+	}
+
+	return &linodeNBConfigs[0], err
+}
+
+// AddNodeToNB adds a backend Node on the Node Balancer configuration
+func AddNodeToNB(
+	ctx context.Context,
+	logger logr.Logger,
+	machineScope *scope.MachineScope,
+	clusterScope *scope.ClusterScope,
+	linodeInstance *linodego.Instance,
+) error {
+	// Update the NB backend with the new instance if it's a control plane node
+	if !kutil.IsControlPlaneMachine(machineScope.Machine) {
+		return nil
+	}
+	// Get the private IP that was assigned
+	addresses, err := machineScope.LinodeClient.GetInstanceIPAddresses(ctx, *machineScope.LinodeMachine.Spec.InstanceID)
+	if err != nil {
+		logger.Error(err, "Failed get instance IP addresses")
+
+		return err
+	}
+	if len(addresses.IPv4.Private) == 0 {
+		err := errors.New("no private IP address")
+		logger.Error(err, "no private IPV4 addresses set for LinodeInstance")
+
+		return err
+	}
+
+	linodeNBConfig, err := GetNodeBalancerConfig(ctx, clusterScope, logger)
```

Comment:
would it be helpful to store the `NodeBalancerConfigID` as a status on the cluster so we can save ourselves a API call for these operations?

Reply:
Yeah, sounds good to me 👍

Reply:
Just to follow-up here, I think more refactoring work is needed to make this change (how we store NB IDs and default the port, potentially needing a defaulting admission webhook for the network config). I've put those changes into a different branch to make a follow-up PR addressing those later on.